lib/workers/repository/index.ts

```diff
@@ -39,15 +42,31 @@ export async function renovateRepository(
     logger.debug('Using localDir: ' + localDir);
     config = await initRepo(config);
     addSplit('init');
-    const { branches, branchList, packageFiles } = await extractDependencies(
-      config
+    const { branches, branchList, packageFiles } = await tracer.startActiveSpan(
+      'extractDependencies',
+      async (span) => {
+        const res = await extractDependencies(config);
+        span.end();
+        return res;
+      }
     );
     if (
       GlobalConfig.get('dryRun') !== 'lookup' &&
       GlobalConfig.get('dryRun') !== 'extract'
     ) {
-      await ensureOnboardingPr(config, packageFiles, branches);
-      const res = await updateRepo(config, branches);
+      await tracer.startActiveSpan('ensure onboarding PR', async (span) => {
+        await ensureOnboardingPr(config, packageFiles, branches);
+        span.end();
+      });
```

Comment:
What about making this into a decorator that could be added to methods to collect telemetry from them? It would be less invasive than this, in my opinion.

I could refactor out the common properties from [here](https://github.com/renovatebot/renovate/blob/main/lib/util/cache/package/decorator.ts) and write some documentation about how to write a custom decorator.

Reply:
I was just thinking similar (that it's quite invasive to be wrapping everything), although I'm not really familiar with decorators so hadn't thought of that as a solution.

Reply:
decorators sounds good to me.

Reply:
Decorators would definitely stream line the process in most cases

Reply:
I extracted the necessary methods from the cache module: https://github.com/renovatebot/renovate/pull/15772

Unfortunately, decorators can only be applied to class methods ☹️ Not ideal, but it's a start.